### PR TITLE
Bump Python

### PR DESCRIPTION
**Pull request by ziw-liu**
Maintainance changes:
- Move from `setup.cfg` to `pyproject.toml`
- Bump Python version to 3.10-3.12

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -5,9 +5,68 @@ build-backend = "setuptools.build_meta"
...
+description = "Wave-optical simulations and deconvolution of optical properties"
+readme = "README.md"
+requires-python = ">=3.10"
+license = { file = "LICENSE" }
+authors = [{ name = "CZ Biohub SF", email = "compmicro@czbiohub.org" }]
+maintainers = [
+    { name = "Talon Chandler", email = "talon.chandler@czbiohub.org" },
+    { name = "Shalin Mehta", email = "shalin.mehta@czbiohub.org" },
+]
+keywords = [
```
Comment (ziw-liu) [inline pyproject.toml:19]: Copied from github repo keywords

File: pyproject.toml [hunk 2]
```diff
@@ -5,9 +5,68 @@ build-backend = "setuptools.build_meta"
...
+    'polarization',
+    'label-free',
+    'permittivity',
+    "reconstruction-algorithm",
+    'qlipp',
+    'mipolscope',
+    'permittivity-tensor-imaging',
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
```
Comment (ziw-liu) [inline pyproject.toml:33]: Now '5 - Production/Stable'?

File: pyproject.toml [hunk 3]
```diff
@@ -5,9 +5,68 @@ build-backend = "setuptools.build_meta"
...
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Image Processing",
+    "Topic :: Scientific/Engineering :: Visualization",
+    "Topic :: Scientific/Engineering :: Information Analysis",
```
Comment (ziw-liu) [inline pyproject.toml:43]: Delete?

File: pyproject.toml [hunk 4]
```diff
@@ -5,9 +5,68 @@ build-backend = "setuptools.build_meta"
 [tool.setuptools_scm]
 write_to = "waveorder/_version.py"
 
+[project]
+name = "waveorder"
+description = "Wave-optical simulations and deconvolution of optical properties"
+readme = "README.md"
+requires-python = ">=3.10"
+license = { file = "LICENSE" }
+authors = [{ name = "CZ Biohub SF", email = "compmicro@czbiohub.org" }]
```
Comment (ziw-liu) [inline pyproject.toml:14]: Change to match CITATION.cff?

File: pyproject.toml (hunk 3 above)
Comment (ziw-liu) [reply pyproject.toml:43]: I meant L42-43.

File: pyproject.toml (hunk 2 above)
Comment (talonchandler) [reply pyproject.toml:33]: I'd leave as `Development Status :: 4 - Beta` for now.

File: pyproject.toml (hunk 3 above)
Comment (talonchandler) [reply pyproject.toml:43]: I agree, thanks.

**Review by talonchandler — DISMISSED** [review]
Thanks for the maintenance, Ziwen!

File: pyproject.toml (hunk 4 above)
Comment (talonchandler) [reply pyproject.toml:14]: Whoops, missed this comment earlier. I think this is fine as is. No need to maintain two lists.

**Review by talonchandler — APPROVED** [review]

**Pull request merged**